docs/composite_image_map_type.js

```diff
@@ -0,0 +1,266 @@
+    img.style.position = 'absolute';
+    // Clear blob from memory as soon as it is loaded.
+    img.onload = () => URL.revokeObjectURL(url);
+    tileDiv.appendChild(img);
+  }
+
+  /**
+   * Removes div from {@link this#tileMap} and sends an event if no tiles remain
+   * to be rendered.
+   * @param {HTMLDivElement} div
+   */
+  markTileCompleted(div) {
+    this.tileMap.delete(div);
+    this.maybeSendEvent(false);
+  }
+
+  /**
+   * If we are transitioning from no pending fetches to pending fetches or vice
+   * versa, send an event.
+   * @param {boolean} adding True if we are about to start a fetch, false if we
+   *     have just finished all fetches
+   */
+  maybeSendEvent(adding) {
+    if (!this.tileMap.size) {
+      const event = new Event('tile');
+      event.count = adding ? 1 : 0;
+      this.eventTarget.dispatchEvent(event);
+    }
+  }
+}
+
+/**
+ * Wrapper object for HTML errors encountered during a single tile fetch, to
+ * avoid throwing errors that would shut down other tile fetches.
+ */
+class ErrorObject {
+  /**
+   * @constructor
+   * @param {number} statusCode HTTP status code (like 404, 400, 429)
+   * @param {string} message Error message from HTTP response
+   * @param {string} url Original URL requested for fetch
+   */
+  constructor(statusCode, message, url) {
+    this.statusCode = statusCode;
+    this.message = message;
+    this.url = url;
+  }
+
+  /**
+   * @return {string} Formatted error message from this error, for console
+   *     display
+   */
+  getMessage() {
+    return 'Could not retrieve ' + this.url +
+        (this.message ? ' (' + this.message + ')' : '');
+  }
+}
+
+ErrorObject.create = (response, url) => {
+  return new ErrorObject(response.status, response.statusText, url);
+};
+
+/**
+ * Performs an HTTP fetch, but with exponential backoff if the server responds
+ * with a "too many requests" error. Return response errors as an ErrorObject
+ * for ease of processing.
+ * @param {string} url URL to fetch
+ * @param {AbortSignal} signal Signal to pass to fetch to trigger abort
+ * @return {Promise<!string|ErrorObject>} Promise with local object URL for
+ *     fetched data, or ErrorObject if an error was encountered, or null if
+ * aborted by signal
+ */
+async function fetchWithBackoff(url, signal) {
+  let retryCount = 0;
+  let response;
+  const exponentialBackoff = new ExponentialBackoff();
+  while (retryCount++ < MAX_RETRIES) {
+    try {
+      response = await fetch(url, {signal: signal});
+      if (response.ok === true) {
+        const blob = await response.blob();
+        return URL.createObjectURL(blob);
+      }
+      if (response.status === TOO_MANY_REQUESTS_STATUS_CODE) {
+        await exponentialBackoff.wait();
+      } else {
+        return ErrorObject.create(response, url);
+      }
+    } catch (err) {
+      if (err.name === 'AbortError') {
+        return null;
+      }
+      throw err;
+    }
+  }
+  return ErrorObject.create(response, url);
```

Comment:
If we got here, it means we exceeded MAX_RETRIES attempts. So we return an error with the code from the latest response (since that's as meaningful as anything else we could do). That allows the caller to log that error, or do whatever they need to.

I don't think returning null here would have the same semantics: currently returning null only happens on abort, when the caller is prepared for null values because they know there was an abort.